metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/_extractors.py

```diff
@@ -0,0 +1,166 @@
+                    "default_schema",
+                    property(new_snowflake_default_schema),
+                )
+            )
+
+            # TODO: Override the BigQuery extractor to use the DataHub SQL parser.
+            # self.extractor_manager.add_extractor()
+
+            yield
+
+    def extract_metadata(
+        self,
+        dagrun,
+        task,
+        complete: bool = False,
+        task_instance=None,
+        task_uuid=None,
+        graph: Optional["DataHubGraph"] = None,
+    ) -> TaskMetadata:
+        self._graph = graph
+        with self._patch_extractors():
+            return super().extract_metadata(
+                dagrun, task, complete, task_instance, task_uuid
+            )
+
+    def _get_extractor(self, task) -> Union[BaseExtractor, None]:
+        extractor = super()._get_extractor(task)
+        if extractor:
+            extractor.set_context(_DATAHUB_GRAPH_CONTEXT_KEY, self._graph)
+        return extractor
+
+
+def _sql_extractor_extract(self: "SqlExtractor") -> TaskMetadata:
+    task_name = f"{self.operator.dag_id}.{self.operator.task_id}"
+    sql = self.operator.sql
+
+    run_facets = {}
+    job_facets = {"sql": SqlJobFacet(query=self._normalize_sql(sql))}
+
+    # Prepare to run the SQL parser.
+    graph = self.context.get(_DATAHUB_GRAPH_CONTEXT_KEY, None)
+
+    database = getattr(self.operator, "database", None)
+    if not database:
+        database = self._get_database()
+
+    # TODO: Add better handling for sql being a list of statements.
+    if isinstance(sql, list):
+        logger.info(f"Got list of SQL statements for {task_name}. Using first one.")
+        sql = sql[0]
+
+    # Run the SQL parser.
+    self.log.debug(
+        "Running the SQL parser (%s): %s",
+        "with graph client" if graph else "in offline mode",
+        sql,
+    )
+    scheme = self.scheme
+    platform = OL_SCHEME_TWEAKS.get(scheme, scheme)
+    sql_parsing_result: SqlParsingResult = create_lineage_sql_parsed_result(
+        query=sql,
+        graph=graph,
+        platform=platform,
+        platform_instance=None,
+        env=builder.DEFAULT_ENV,
+        database=database,
+        schema=self.default_schema,
+    )
+    self.log.debug(f"Got sql lineage {sql_parsing_result}")
+
+    if sql_parsing_result.debug_info.error:
+        error = sql_parsing_result.debug_info.error
+        run_facets["extractionError"] = ExtractionErrorRunFacet(
+            totalTasks=1,
+            failedTasks=1,
+            errors=[
+                ExtractionError(
+                    errorMessage=str(error),
+                    stackTrace=None,
+                    task="datahub_sql_parser",
+                    taskNumber=None,
+                )
+            ],
+        )
+
+    # Save sql_parsing_result to a fake facet.
+    run_facets["datahub_sql"] = sql_parsing_result
+
+    return TaskMetadata(
+        name=task_name,
+        inputs=[],
+        outputs=[],
```

Comment:
Do we gain anything by setting this, e.g. in the airflow UI?

Reply:
nope - TaskMetadata is an OL thing, not an Airflow thing